stan/math/prim/scal/meta/is_index.hpp

```diff
@@ -0,0 +1,19 @@
+#ifndef STAN_MATH_PRIM_SCAL_META_IS_INDEX_HPP
+#define STAN_MATH_PRIM_SCAL_META_IS_INDEX_HPP
+
+#include <stan/math/prim/scal/meta/bool_constant.hpp>
+#include <type_traits>
+
+namespace stan {
+
+/**
+ * Deduces whether type is non-floating point arithmetic
```

Comment:
This doc doesn't really convey the purpose of the type trait

What's the ideal use-case for this? Is it different from ```is_integer```?

Reply:
> What's the ideal use-case for this? Is it different from is_integer?

Lol I'm dumb, `is_integer is exactly what I need so I can delete `is_index`

Side note: